devops/providers/azure-devops/templates/app_service_maven/README.md

```diff
@@ -0,0 +1,77 @@
+        | Unit Tests | `build_stage.yml` | Automatically detects and runs unit tests using maven tasks. |
+        | Archive Integration tests | `build_stage.yml` | Zips integration tests. |
+
+    - **[deploy-stages.yml](./deploy-stages.yml)**
+
+        Per cloud provider, the `deploy-stage.yml` executes a set of tasks for every environment passed to it from a **Maven Service** `yaml` file. The final task in this file runs integration tests.
+
+        | primary features | deploy stage file | behavior |
+        | ---  | ---   | ---  |
+        | Cloud Based Deployments | `deploy-stage.yml` | This file enables the deploy steps to run per cloud provider. Our implementation is Azure bound. |
+        | Environment Based Deployments  | `deploy-stage.yml` | This file enables the deploy steps to run per a cloud provider's configured list of environments. |
+        | Detect App Service Jar File | `app-service-detect-jar.yml` | Scans the incoming artifact drop folder for the jar file. |
+        | Publish Jar File Image | `app-service-acr-publish.yml` | Uploads jar file to image container registry. |
+        | [Maven Deploy](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/maven?view=azure-devops) | `app-service-deployment-steps.yml` | Deploy application using maven tasks. |
+        | App Service CL Update | `app-service-update-commandline.yml` | Updates the App Service startup command with information about the newly deployed JAR file. |
+        | Integration Tests | `app-service-deployment-steps.yml` | Automatically detects and runs integration tests using maven tasks. |
+
+- ### Variable group naming conventions
+
+    Variable group naming conventions should be respected. They are hardcoded in the following `yaml` files. It's not recommended to change variable group names.
+
+    | Variable Group | YAML FILE |
+    | ---      | ---         |
+    |  `Azure - Common` | Service yaml |
+    |  `Azure - Common` | deploy-stages.yml |
+    |  `${{ provider.name }} Target Env - ${{ environment }}` | deploy-stages.yml |
+    |  `${{ provider.name }} Target Env Secrets - ${{ environment }}` | deploy-stages.yml |
+    |  `${{ provider.name }} Service Release - ${{ parameters.serviceName }}` | deploy-stages.yml |
+
+- ### Cloud provider boundaries
+
+    The **Shared Maven Service Pipeline** currently accomodates a multi-cloud **Maven Service** deployment. However, current implementation is Azure bound. Azure bound means that if you have a multi-cloud Maven Service, this pipeline only has an execution workflow targeting Azure infrastructure. The service contracts for other cloud providers are in place but have not been implemented.  In short, deployments to a cloud provider are bound by their `yaml` pipeline configuration, the variable groups that belong to them and whether or not the Maven Service solution includes that cloud provider's implementation.
+
+## PR vs Master builds in AzDO
```

Comment:
This is actually a feature of the *consuming* code - i.e., the actual service pipeline that uses these templates. The template does not actually do this for you.

Reply:
Glad you caught this. This will be removed.